### PR TITLE
Add redux actions, reducers, and tests

### PR DESCRIPTION
**Pull request by rianadon**
I still have a lot of tests to write, but decided to PR early for feedback.

My current setup is that redux only tracks one state variable: `graph`, which accounts for the current iok graph. The format of `graph` is the same as the files we import. The actions correspond to what our toolbar buttons will do (adding nodes, editing them, deleting, and merging). Merging has yet to be implemented.

I've also yet to integrate Redux with components. Lots to do :grin: 

And the `storage` files are changed because I realized eslint wasn't linting subfolders, so these files never got linted.

### CODE REVIEW

File: v4/src/redux/actions.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export const addNode = (parentId, props) => ({
+  type: 'ADD_NODE',
```
Comment (rustielin) [inline v4/src/redux/actions.js:2]: nit: add these to TYPES

File: v4/src/redux/reducers/graph.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      newNode.shift(oldNode.position());
+      oldNode.remove();
+      break;
+    }
+    case 'DELETE_NODE': {
+      cy.getElementById(action.id).remove();
+      break;
+    }
+    case 'MERGE_NODE': {
+      console.log("Huh? I can't hear you, honey. A little louder please?");
```
Comment (rustielin) [inline v4/src/redux/reducers/graph.js:61]: nit: use our logging module or you could alert() on this lol. Love these easter eggs :)

File: v4/src/redux/reducers/graph.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import Cytoscape from 'cytoscape';
+import { sha256 } from 'js-sha256';
+
+function edgeId(sourceId, targetId) {
```
Comment (rustielin) [inline v4/src/redux/reducers/graph.js:4]: what about node IDs?

**Comment by rustielin** [comment]
Everything looks pretty great. Just a few nits here and there. Love your code style and general cleanliness of everything. Looking forward to getting this merged in :p Also thanks for catching the issue with linting subfolders

File: v4/src/redux/actions.js (hunk 1 above)
Comment (rianadon) [reply v4/src/redux/actions.js:2]: Do you mean something like:
```javascript
export const ADD_NODE = 'ADD_NODE';
```

Comment (rustielin) [reply v4/src/redux/actions.js:2]: Something like this: https://github.com/rustielin/iok/blob/b7a55fa18fcdfd9ab16b4ffb5fa6212a0f01a053/v4/src/types.js#L4 for each of the redux action types

File: v4/src/redux/reducers/graph.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  };
+}
+
+function edgeId(sourceId, targetId) {
+  const hash = sha256.create();
+  hash.update(sourceId + targetId);
+  return hash.hex();
+}
+
+function createEdge(source, target) {
```
Comment (rustielin) [inline v4/src/redux/reducers/graph.js:32]: nit: probably worth documenting the type of `source` and `target

File: v4/src/redux/reducers/graph.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  // create new node, a mix of nodes From and To
+  const newNode = cy.add(createNode({
+    name: `${from.data('name')} || ${to.data('name')}`,
+    node_type: NTYPE.TOPIC,
+  }));
+  const nodes = {};
+
+  // if the node is repeated / in hashtable, merge it, else, add i
+  // eslint-disable-next-line no-restricted-syntax
+  for (const node of incomers(from).union(incomers(to))) {
```
Comment (rustielin) [inline v4/src/redux/reducers/graph.js:64]: nice!

**Pull request merged**